provider/in_place_upgrade_base.py

```diff
@@ -0,0 +1,172 @@
+import os
+import re
+import time
+import logging
+
+
+from virttest import utils_package
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class IpuTest(object):
+    """
+    Class for in_place_upgrade test in the vm
+    """
+
+    def __init__(self, test, params, env):
+        """
+        Init the default values of in_place_upgrade object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.session = None
+        self.test = test
+        self.params = params
+        self.env = env
+
+    def run_guest_cmd(self, cmd, check_status=True, timeout=1200):
+        """
+        Run command in guest
+
+        :param cmd: A command needed to run
+        :param check_status: If true, check the status after running the cmd
+        :return: The output after running the cmd
+        """
+        status, output = self.session.cmd_status_output(cmd, timeout=timeout)
+        if check_status and status != 0:
+            self.test.fail("Execute command %s failed, output: %s"
+                           % (cmd, output))
+        return output.strip()
+
+    def upgrade_process(self, cmd, timeout=6000):
+        """
+        Fix known issues before doing in place upgradet
+
+        :param cmd: A command needed to run
+        """
+        self.session.cmd(cmd, timeout=timeout)
+
+    def yum_update_no_rhsm(self, test, old_custom):
+        """
+        do yum update in the vm
+        """
+        try:
+            pkgs = self.params.get("depends_pkgs").split()
+            if not utils_package.package_install(pkgs, self.session):
+                test.cancel("Install dependency packages failed")
+            self.session.cmd(old_custom)
+            dis_content = self.params.get("disable_content")
+            self.session.cmd(dis_content, timeout=300)
+            ena_content = self.params.get("enable_content")
+            self.session.cmd(ena_content, timeout=300)
+            if os.path.exists("/var/run/yum.pid"):
```

Comment:
This checks the file in the host, but the yum command is running in the guest.

Reply:
Updated it.